### PR TITLE
New module: Add monitoring/zabbix_proxy module.

### PR DESCRIPTION
**Pull request by akomic**
##### SUMMARY
Implements zabbix_proxy module to manipulate zabbix proxy objects on zabbix server via API.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
modules/monitoring/zabbix_proxy

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel bf322c0970) last updated 2017/05/16 12:43:20 (GMT +200)
  config file = /Users/ak/.ansible.cfg
  configured module search path = [u'/Users/ak/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/ak/Stuff/ansible-zabbix/ansible/lib/ansible
  executable location = /Users/ak/Stuff/ansible-zabbix/ansible/bin/ansible
  python version = 2.7.13 (default, Dec 17 2016, 23:03:43) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (gundalow) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: Is this line right, the dates look fairly old

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (gundalow) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: Please remove all `version_added:` under `options:`

:	
Only needed if this option was extended after initial Ansible release, i.e. this is greater than the top level version_added field.

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (gundalow) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: `U(https://www.zabbix.com/documentation/3.2/manual/appendix/api/hostinterface/definitions#host_interface)`  will generate a hyperlink

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (gundalow) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#formatting-options

**Review by gundalow — CHANGES_REQUESTED** [review]

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
@akomic Thanks for addressing my comments. Need to find one other person to review this before we can get it merged.

@eikef I believe you are a maintainer for the other Zabbix modules, it would be good to get your input on this. Thanks in advance :)

**Comment by ttiops** [comment]
Is anything else required @eikef @gundalow ?

**Comment by akomic** [comment]
Is anything else required @eikef @gundalow ?

**Comment by gundalow** [comment]
Closing and reopening to rerun CI

**Comment by akomic** [comment]
Any updates on this one? Anything required from my side?

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: metadata_version should safely be updatable to 1.1 for this new module :)

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: Did you mean to write "Zabbix host" instead of "Zabbix proxy"?

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: 2.5 now

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: Is this meant to say host instead of proxy?

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_proxy.py:133]: 5 is not documented as a valid value above (this is the internal value of the Zabbix API).

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (eikef) [inline lib/ansible/modules/monitoring/zabbix_proxy.py]: `final_interface = dict((k,str(v)) for k,v in final_interface.iteritems())`

seems to be required here since the values from new_interface may contain non-strings while the values returned by the API are always strings. With this added line, the module becomes idempotent with passive proxies.

**Review by eikef — COMMENTED** [review]
This looks good to me and it worked in tests (other than the idempotency for interface values). With those changes made I'd vote to ship it.

File: lib/ansible/modules/monitoring/zabbix_proxy.py
Comment (akomic) [reply lib/ansible/modules/monitoring/zabbix_proxy.py:133]: This should be fixed with the new commit. Take a look.

**Comment by eikef** [comment]
shipit

**Comment by eikef** [comment]
Thank you @akomic, looks good :-)

**Comment by resmo** [comment]
shipit

**Pull request merged**
